lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -1224,6 +1224,41 @@ def group_log_entries_by_tag_and_local_resource_id(chunk)
     # collection of entries.
     def determine_group_level_monitored_resource_and_labels(tag,
                                                             local_resource_id)
+      # Determine the monitored resource based on the local_resource_id.
+      # Different monitored resource types have unique ids in different format.
+      # We will query Metadata Agent for the monitored resource. Return the
+      # legacy monitored resource (either the instance resource or the resource
+      # inferred from the tag) if failed to get a monitored resource from
+      # Metadata Agent with this key.
+      #
+      # Examples:
+      # "container.<container_id>" // Docker container.
+      # "k8s_pod.<namespace_name>.<pod_name>" // GKE pod.
+      if local_resource_id
+        resource = monitored_resource_from_local_resource_id(local_resource_id)
+        if resource
+          resource.freeze
+          resource.labels.freeze
+
+          common_labels = determine_common_labels_for_resource_type(
+            resource.type)
+
+          # TODO(bmoyles0117): Remove this once backend service can map metadata
+          # labels.
+          if resource.type == GKE_CONSTANTS[:resource_type]
+            common_labels.merge!(
```

Comment:
`delete_and_extract_labels` need to extract `namespace_name` and `pod_name` from a hash that include these two fields (in the legacy path it's from the regex matched group). Here only the `resource.labels` have those fields. However, the `delete_and_extract_labels` method also *delete* the specified fields as a side effect.

I would recommend just doing it explicitly.
```
            common_labels.merge!(
               'namespace_name' => resource.labels['namespace_id'],
               'pod_name' => resource.labels['pod_id']
            )
```

Reply:
Thanks, updated